src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/Visitors/DictionaryObjectTypeVisitor.cs

```diff
@@ -79,7 +79,7 @@ public override void Visit(IAcceptor acceptor, KeyValuePair<string, Type> type,
                 var reference = new OpenApiReference()
                 {
                     Type = ReferenceType.Schema,
-                    Id = underlyingType.GetOpenApiReferenceId(isDictionary: false, isList: false, namingStrategy)
+                    Id = underlyingType.GetOpenApiReferenceId(underlyingType.IsOpenApiDictionary(), underlyingType.IsOpenApiArray(), namingStrategy)
```

Comment:
In addition to this change, you should find the same fix on `ListObjectTypeVisitor` as well.